pkg/tnf/handlers/podantiaffinity/podantiaffinity.json

```diff
@@ -0,0 +1,34 @@
+{
+  "identifier": {
+    "url": "http://test-network-function.com/tests/podantiaffinity",
+    "version": "v1.0.0"
+  },
+  "description": "This test checks cnf pod antiaffinity rule in cnf deployment.",
+  "testResult": 0,
+  "testTimeout": 10000000000,
+  "reelFirstStep": {
+    "execute": "oc get deployment {{.DEPLOYMENT_NAME}} -n {{.DEPLOYMENT_NAMESPACE}} -o json | jq -r ' if .spec.replicas > 1 then if .spec.template.spec.affinity.podAntiAffinity != null then \"OK\" else \"PODAAFAIL\" end else \"REPLICAFAIL\" end '\n",
+    "expect": [
+      "(?m)OK",
+      "(?m)PODAAFAIL",
+      "(?m)REPLICAFAIL"
+    ],
+    "timeout": 10000000000
+  },
+  "resultContexts": [
+    {
+      "pattern": "(?m)OK",
+      "defaultResult": 1
```

Comment:
I see failure logs like this right now:

  Expected
        <int>: 2
    to equal
        <int>: 1

It doesn't show what result was causing code 2. @HamadiSE, can we enhance the implementation for json based test and make it print the actual pattern matched instead of just the code?

Reply:
I think so. There's the generic.go that handles this and can be extended.
Also, with the current deployment files, this test will fail because we don't have replicas or anti affinity defined.

Reply:
my comment get posted twice, then when I delete it, it gets deleted from both places -I need to get familiar with this github better-

regarding the comments, we can use the generic.go to be more verbose in case of failure, we can print the expected and the matched pattern

Reply:
https://issues.redhat.com/browse/CNFCERT-46 created

Reply:
Thanks. @wying3 Make the error string/pattern more readable and descriptive. Once the task Salah created is done, it will show up in the log.

Reply:
@wying3 can you change "PODAAFAIL" and "REPLICAFAIL" to something more descriptive?

Reply:
@wying3 I guess you missed this comment?

Reply:
ok, will make changes

Reply:
how about change to ANTIAFFINITYMISSING  and REPLICAEQ1?

Reply:
new match pattern: replica count is 1 and Antiaffinity missing